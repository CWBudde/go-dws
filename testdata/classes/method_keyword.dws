// Test file demonstrating that 'method' keyword works identically to 'function'/'procedure'
// for declaring class instance methods

type TPoint = class
  X, Y: Integer;

  // Using 'method' keyword with return type (like 'function')
  method GetX(): Integer;
  begin
    Result := X;
  end;

  // Using 'method' keyword without return type (like 'procedure')
  method SetX(value: Integer);
  begin
    X := value;
  end;

  // Using traditional 'function' keyword (should still work)
  function GetY(): Integer;
  begin
    Result := Y;
  end;

  // Using traditional 'procedure' keyword (should still work)
  procedure SetY(value: Integer);
  begin
    Y := value;
  end;

  // Constructor using traditional syntax
  constructor Create(aX, aY: Integer);
  begin
    X := aX;
    Y := aY;
  end;
end;

var p := new TPoint(10, 20);

PrintLn('Initial values:');
PrintLn('X = ' + IntToStr(p.GetX()));
PrintLn('Y = ' + IntToStr(p.GetY()));

p.SetX(30);
p.SetY(40);

PrintLn('After setting new values:');
PrintLn('X = ' + IntToStr(p.GetX()));
PrintLn('Y = ' + IntToStr(p.GetY()));
