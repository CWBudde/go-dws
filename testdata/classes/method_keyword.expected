Initial values:
X = 10
Y = 20
After setting new values:
X = 30
Y = 40
